### PR TITLE
EBNT-300 pipelines prototype

### PR DESCRIPTION
**Pull request by mike0sv**

### CODE REVIEW

File: examples/pipeline_example/run_instance.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    p = s1.as_pipeline().append(s2)
+
+    ebnt = Ebonite.local(clear=True)
+    task = ebnt.get_or_create_task('a', 'b')
+    task.push_model(s1)
+    task.push_model(s2)
+    task.add_pipeline(p)
+    with use_local_installation():
+        image = ebnt.build_image('ebnt_pipeline_test', p, task, FlaskServer(), force_overwrite=True)
+        instance = ebnt.run_instance('ebnt_pipeline_test', image, server=FlaskServer())
```
Comment (i-a-andrianov) [inline examples/pipeline_example/run_instance.py:27]: Why do we need to specify server for running? I think this setting is relevant for building only.

File: tests/repository/metadata/meta_common.py [hunk 2]
```diff
@@ -920,6 +1223,7 @@ def test_update_environment__not_existing(meta: MetadataRepository, environment)
 
 
 def test_delete_environment__ok(meta: MetadataRepository, created_environment):
+    created_environment
```
Comment (i-a-andrianov) [inline tests/repository/metadata/meta_common.py:1226]: Looks like a coding artifact

File: examples/pipeline_example/run_instance.py (hunk 1 above)
Comment (mike0sv) [reply examples/pipeline_example/run_instance.py:27]: oh, thats a brainfart))))

File: src/ebonite/client/base.py [hunk 3]
```diff
@@ -79,18 +80,14 @@ def push_model(self, model: Model, task: Task = None) -> Model:
         model = self.meta_repo.save_model(model)
         return model
 
-    def delete_model(self, model: Model, *, force=False, cascade=False):
```
Comment (ElmiRakh) [inline src/ebonite/client/base.py:82]: Are we getting rid of cascade deletions?

File: src/ebonite/repository/metadata/base.py [hunk 4]
```diff
@@ -338,12 +341,107 @@ def save_model(self, model: Model) -> Model:
...
+
+        pass  # pragma: no cover
+
+    @abstractmethod
+    def create_pipeline(self, pipeline: Pipeline) -> Pipeline:
+        """
+        Creates model in the repository
+
+        :param pipeline: pipeline to create
+        :return: created model
```
Comment (ElmiRakh) [inline src/ebonite/repository/metadata/base.py:389]: :return: created model - probably pipeline?

File: src/ebonite/client/base.py [hunk 5]
```diff
@@ -144,42 +145,59 @@ def get_model(self, model_name: str, task: TaskVar, project: ProjectVar = None,
...
+        """
+        Load pipeline from repository
+
+        :param name: pipeline name to load
+        :param task: :py:class:`~ebonite.core.objects.Task` instance to load image from
+        :return: loaded :py:class:`~ebonite.core.objects.Pipeline` instance
+        """
+        return self.meta_repo.get_pipeline_by_name(name, task)
+
+    def delete_pipeline(self, pipeline: Pipeline):
```
Comment (ElmiRakh) [inline src/ebonite/client/base.py:187]: shouldn't it have cascade parameter and related functionality to remove associated models?

Comment (mike0sv) [reply src/ebonite/client/base.py:187]: no)))

**Pull request merged**